commit c7ea1c80d6d553e45bb5ca17fea3eef73494fc95
Author: Sergey Yanovich <ynvich@gmail.com>
Date:   Thu Apr 17 01:47:09 2008 +0300

    Drop keycode bitmap check in input events
    
    Debian and Ubuntu rely on a simple acpi_fakekey program to convey
    user events collected from ACPI to handlers. That program just
    writes to the default keyboard event device. Thus it allows hadlers
    to concentrate on a single source of events - keyboard.
    
    Not every keyboard has all buttons, which can be mapped. And not all
    existing buttons are reported as present, even. So commit
    8006479c9b75fb6594a7b746af3d7f1fbb68f18f seem to break hotkeys on
    many laptops. This patch removes the breaking check.

diff --git a/drivers/input/input.c b/drivers/input/input.c
index f02c242..7bdd8c5 100644
--- a/drivers/input/input.c
+++ b/drivers/input/input.c
@@ -159,8 +159,7 @@ static void input_handle_event(struct input_dev *dev,
 		break;
 
 	case EV_KEY:
-		if (is_event_supported(code, dev->keybit, KEY_MAX) &&
-		    !!test_bit(code, dev->key) != value) {
+		if (!!test_bit(code, dev->key) != value) {
 
 			if (value != 2) {
 				__change_bit(code, dev->key);
